src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportRelatedObjectsResource.kt

```diff
@@ -41,17 +42,17 @@ abstract class ReportRelatedObjectsResource<ResponseDto, AddRequest, UpdateReque
       ?: throw ReportNotFoundException(this)
   }
 
-  protected fun <T> (UUID).updateReportOrThrowNotFound(changeMessage: String, block: (Report) -> T): T {
+  protected fun <T> (UUID).updateReportOrThrowNotFound(changeMessage: String, whatChanged: WhatChanged? = null, block: (Report) -> T): T {
     val report = findReportOrThrowNotFound()
     return block(report).also {
       report.modifiedAt = LocalDateTime.now(clock)
       report.modifiedBy = authenticationFacade.getUserOrSystemInContext()
 
       val basicReport = report.toDtoBasic()
       eventPublishAndAudit(
-        // TODO: should different related object actions raise different events?
```

Comment:
@ushkarev I think this should address this TODO so [I've removed it](https://github.com/ministryofjustice/hmpps-incident-reporting-api/pull/129/commits/d952ee3eaffe2a06717488e7b2eb1236e35c8eee).

I believe this is a good middle ground and better than having a lot of possible event types for each part of the Report that may or may have not changed.

The result should be similar: Consumer knows that a Report has been updated and what part (roughly) changed - but it's less unwieldy than having another 10 or so domain event types ☺️